File: fe/src/submodules/tracker/types/initiative.ts
export interface TrackerInitiative {
  id: string
  product_id: string
  key: string
  name: string
  description: string | null
  status: string
  owner_id: string | null
  target_date: string | null
  created_at: string
  updated_at: string
  deleted_at: string | null
}

export interface CreateInitiativePayload {
  product_id: string
  key: string
  name: string
  description?: string | null
  status?: string
  owner_id?: string | null
  target_date?: string | null
}

export interface UpdateInitiativePayload {
  key?: string
  name?: string
  description?: string | null
  status?: string
  owner_id?: string | null
  target_date?: string | null
  deleted_at?: string | null
}

File: be/app/submodules/tracker/models/initiative.py
"""TrackerInitiative model."""

from __future__ import annotations

from datetime import UTC, date, datetime

from beanie import DocumentWithSoftDelete, PydanticObjectId
from pydantic import Field


def _utcnow() -> datetime:
    return datetime.now(UTC)


class TrackerInitiative(DocumentWithSoftDelete):
    product_id: PydanticObjectId
    key: str
    name: str
    description: str | None = None
    status: str = "planned"
    owner_id: PydanticObjectId | None = None
    target_date: date | None = None
    created_at: datetime = Field(default_factory=_utcnow)
    updated_at: datetime = Field(default_factory=_utcnow)

    class Settings:
        name = "tracker_initiatives"
        indexes = ["product_id", "key", "status", "owner_id"]

    async def touch(self) -> None:
        self.updated_at = _utcnow()
        await self.save()
